src/hotspot/share/oops/method.hpp

```diff
@@ -725,7 +726,9 @@ class Method : public Metadata {
   // (see AsyncGetCallTrace support for Forte Analyzer) and this
   // needs to be async-safe. No allocation should be done and
   // so handles are not used to avoid deadlock.
-  jmethodID find_jmethod_id_or_null()               { return method_holder()->jmethod_id_or_null(this); }
+  jmethodID find_jmethod_id_or_null()               {
+    return method_holder() != nullptr ? method_holder()->jmethod_id_or_null(this) : nullptr;
```

Comment:
If `method_holder()` is null at this point what does that mean for the lifecycle of the Method?

Reply:
Please, ignore this part of code for the time being. I had a crash in CI which was pointing vaguely to this code - unfortunately, the hs_err.log files are not preserved in the test archives and I am not able to reproduce the failure locally. I need to debug the crash and make sure I understand the root cause.

Reply:
_Update:_ I was able to get to the bottom of the methods not having method holder associated with them.

The `ClassFileParser` does not finalize initialization of the `InstanceKlass` it has created if `_klass != nullptr` (https://github.com/openjdk/jdk/blob/9727f4bdddc071e6f59806087339f345405ab004/src/hotspot/share/classfile/classFileParser.cpp#L5161). This also means, that the `Method` instances are not wired to their method holders via 'constant method'->'constant pool'->'pool holder' chain. However, they need to be deallocated and as such I really need a distinguishing argument for `InstanceKlass::deallocate_methods` call such that I don't attempt to resolve `jmethodid` values in that case.